client/components/Dashboard.tsx

```diff
@@ -31,47 +29,243 @@ const GET_PROJECT = gql`
+                time: '',
+                complexity: '',
+            });
+        } else {
+            setOrder(true);
+            setArrow({
+                ...arrow,
+                depth: '↓',
+                time: '',
+                complexity: '',
+            });
+        }
+        if (order) {
+            const newArr = [];
+            if (selectedProject) {
+                // eslint-disable-next-line no-restricted-syntax, guard-for-in
+                for (const key in selectedProject?.queries) {
+                    newArr.push(selectedProject?.queries[key]);
+                }
+            }
+            const dataToSort = [...newArr];
+            dataToSort.sort((a, b) => a.depth - b.depth);
+            selectQuerries(dataToSort);
+        } else {
+            const newArr = [];
+            if (selectedProject) {
+                // eslint-disable-next-line no-restricted-syntax, guard-for-in
+                for (const key in selectedProject?.queries) {
+                    newArr.push(selectedProject?.queries[key]);
+                }
+            }
+            const dataToSort = [...newArr];
+            dataToSort.sort((a, b) => b.depth - a.depth);
+            selectQuerries(dataToSort);
+        }
+    };
+    const sortByComplexity = (): any => {
+        if (order === true) {
+            setOrder(false);
+            setArrow({
+                ...arrow,
+                complexity: '↑',
+                time: '',
+                depth: '',
+            });
+        } else {
+            setOrder(true);
+            setArrow({
+                ...arrow,
+                complexity: '↓',
+                depth: '',
+                time: '',
+            });
+        }
+        if (order) {
+            const newArr = [];
+            if (selectedProject) {
+                // eslint-disable-next-line no-restricted-syntax, guard-for-in
+                for (const key in selectedProject?.queries) {
+                    newArr.push(selectedProject?.queries[key]);
+                }
+            }
+            const dataToSort = [...newArr];
+            dataToSort.sort((a, b) => a.complexity - b.complexity);
+            selectQuerries(dataToSort);
+        } else {
+            const newArr = [];
+            if (selectedProject) {
+                // eslint-disable-next-line no-restricted-syntax, guard-for-in
+                for (const key in selectedProject?.queries) {
+                    newArr.push(selectedProject?.queries[key]);
+                }
+            }
+            const dataToSort = [...newArr];
+            dataToSort.sort((a, b) => b.complexity - a.complexity);
+            selectQuerries(dataToSort);
+        }
+    };
+    useEffect(() => {
+        // if (!loadingR && dataR) {
+        //     setProjects(dataR.user.projects);
+        // }
         if (!loading && data) {
             selectProject(data.project);
+            selectQuerries(selectedProject?.queries);
         }
```

Comment:
I'm not familiar with how loading is updated by the `useQuery` hook imported from apollo client. Does a cleanup function need to be returned from the `useEffect` to set `loading` to false when the component unmounts?

Reply:
Loading is true up until the data is resolved. One the data is resolved and fetched, the loading becomes falsy.